index.js

```diff
@@ -1,5 +1,74 @@
 'use strict';
 
+var colorConvert = require('color-convert');
+var supportsColor = require('supports-color');
+
+supportsColor = supportsColor || {};
+supportsColor.level = supportsColor.level || 0;
+
+colorConvert.rgb.rgb = function rgb (rgb) {
+	return rgb;
+};
+
+function getConverter (color) {
+	var has = function has (name) {
+		return color.hasOwnProperty(name);
+	}
+	switch (true) {
+		case has('r') && has('g') && has('b'): return 'rgb';
+		case has('h') && has('s') && has('l'): return 'hsl';
+		case has('h') && has('s') && has('v'): return 'hsv';
+		case has('h') && has('w') && has('b'): return 'hsv';
+		case has('c') && has('m') && has('y') && has('k'): return 'cmyk';
+		case has('css'): return 'keyword';
+		default: return null;
+	}
+}
+
+function getArrayFromKeyString(obj, str) {
+	if (str === 'keyword') {
+		return [obj['css']];
+	}
+
+	return str.split('').map(function(key) {
+		return obj[key];
+	});
+}
+
+var levels = [
+	function level0 () {
+		return '';
+	},
+	function level1 (color, offset) {
+		var converter = getConverter(color);
+		if (!converter) {
+			return '';
+		}
+		var args = getArrayFromKeyString(color, converter);
+		var code = colorConvert[converter]['ansi16'](args);
+		return '\u001b[' + (code + offset) + 'm';
+	},
+	function level2 (color, offset) {
+		var converter = getConverter(color);
+		if (!converter) {
+			return '';
+		}
+		var args = getArrayFromKeyString(color, converter);
+		var code = colorConvert[converter]['ansi'](args);
+		return '\u001b[' + (38 + offset) + ';5;' + code + 'm';
+	},
+	function level3 (color, offset) {
+		var converter = getConverter(color);
+		if (!converter) {
+			return '';
+		}
+		var args = getArrayFromKeyString(color, converter);
+		var code = colorConvert[converter]['rgb'](args);
+		return '\u001b[' + (38 + offset) + ';2;' +
+			code[0] + ';' + code[1] + ';' + code[2] + 'm';
+	}
+];
```

Comment:
The last three methods in this array seem, besides the last line, all pretty much exact. Let's try to keep it DRY?